packages/core/useTransition/index.test.ts

```diff
@@ -1,146 +1,230 @@
+    const source = ref(0)
     const duration = ref(100)
+    const transition = useTransition(source, { duration })
 
-    const transitionedValue = useTransition(baseValue, {
-      duration,
-      transition: n => n,
-    })
+    source.value = 1
 
-    // first transition should take 100ms
-    baseValue.value = 1
+    await promiseTimeout(50)
+    expectBetween(transition.value, 0, 1)
 
-    await promiseTimeout(150)
-    expect(transitionedValue.value).toBe(1)
+    await promiseTimeout(100)
+    expect(transition.value).toBe(1)
 
-    // second transition should take 200ms
     duration.value = 200
-    baseValue.value = 2
+    source.value = 2
 
     await promiseTimeout(150)
-    expect(transitionedValue.value < 2).toBe(true)
+    expectBetween(transition.value, 1, 2)
 
     await promiseTimeout(100)
-    expect(transitionedValue.value).toBe(2)
+    expect(transition.value).toBe(2)
   })
 
-  it('calls onStarted and onFinished callbacks', async() => {
+  it('fires onStarted and onFinished callbacks', async() => {
+    const source = ref(0)
     const onStarted = jest.fn()
     const onFinished = jest.fn()
 
-    const baseValue = ref(0)
-
-    useTransition(baseValue, {
+    useTransition(source, {
       duration: 100,
-      onFinished,
       onStarted,
-      transition: n => n,
+      onFinished,
     })
 
     expect(onStarted).not.toHaveBeenCalled()
     expect(onFinished).not.toHaveBeenCalled()
 
-    baseValue.value = 1
-    await nextTick()
+    source.value = 1
 
+    await promiseTimeout(50)
     expect(onStarted).toHaveBeenCalled()
     expect(onFinished).not.toHaveBeenCalled()
 
+    onStarted.mockReset()
+    onFinished.mockReset()
+
+    await promiseTimeout(100)
+    expect(onStarted).not.toHaveBeenCalled()
+    expect(onFinished).toHaveBeenCalled()
+  })
+
+  it('clears pending transitions before starting a new one', async() => {
+    const source = ref(0)
+    const onStarted = jest.fn()
+    const onFinished = jest.fn()
+
+    useTransition(source, {
+      delay: 100,
+      duration: 100,
+      onFinished,
+      onStarted,
+    })
+
+    await promiseTimeout(150)
+    expect(onStarted).not.toHaveBeenCalled()
+    source.value = 1
+    await promiseTimeout(50)
+    source.value = 2
+    await promiseTimeout(250)
+    expect(onStarted).toHaveBeenCalledTimes(1)
+    expect(onFinished).toHaveBeenCalledTimes(1)
+  })
+
+  it('can be disabled for sychronous changes', async() => {
+    const onStarted = jest.fn()
+    const disabled = ref(false)
+    const source = ref(0)
+
+    const transition = useTransition(source, {
+      disabled,
+      duration: 100,
+      onStarted,
+    })
+
+    disabled.value = true
+    source.value = 1
+
+    expect(transition.value).toBe(1)
     await promiseTimeout(150)
-    expect(onStarted.mock.calls.length).toBe(1)
-    expect(onFinished.mock.calls.length).toBe(1)
+    expect(onStarted).not.toHaveBeenCalled()
+    disabled.value = false
+    expect(transition.value).toBe(1)
```

Comment:
Sorry for the messy diff here. I tried to keep this clean by deleting the old test file in it's own commit, but apparently that did not matter. The only real difference here is the addition of the disabled test.